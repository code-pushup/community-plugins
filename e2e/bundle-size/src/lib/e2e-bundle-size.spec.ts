import {afterEach, beforeEach, describe, expect, it} from "vitest";
import {
    addCodePushupConfigFile, cleanTestFolder,
    executeCliCollectForPlugin,
    getRelativePathToPluginDist, omitVariableReportData,
    readReportJson
} from "testing-utils";
import {resolve} from "path";
import {Report, reportSchema} from "@code-pushup/models";

describe('execute-plugin-bundle-size', () => {
    const baseDir = resolve('tmp/e2e/execute-plugin-bundle-size');

    beforeEach(async () => {
        await addCodePushupConfigFile( {
            fileImports: [`import {bundleSizePlugin} from "${getRelativePathToPluginDist(baseDir, 'bundle-size')}";`],
            pluginStrings: ['bundleSizePlugin()'],
            targetFolder: baseDir
        });
    });

    afterEach(async () => {
      await cleanTestFolder(baseDir);
    });

    it('should run bundle-size plugin and create valid report.json', async () => {
      const { code, stderr } = await executeCliCollectForPlugin('bundle-size', {}, baseDir);
      expect(code).toBe(0);

        const report: Report = await readReportJson(baseDir);
        expect(() => reportSchema.parse(report)).not.toThrow();
        expect(omitVariableReportData(report)).toMatchSnapshot();
    });

    it('should run bundle-size plugin with the correct slug and audits in report.json', async () => {
        const { code, stderr } = await executeCliCollectForPlugin('bundle-size', {}, baseDir);
        expect(code).toBe(0);

        const report: Report = await readReportJson(baseDir);
        expect(report.plugins.at(0)?.slug).toBe('bundle-size');
        expect(report.plugins.at(0)?.audits.length).toBe(1);
        expect(report.plugins.at(0)?.audits.at(0)?.slug).toBe('bundle-size');

    });
});
