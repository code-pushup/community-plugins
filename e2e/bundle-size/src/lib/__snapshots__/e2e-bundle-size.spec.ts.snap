// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`execute-plugin-bundle-size > should run bundle-size plugin and create valid report.json 1`] = `
{
  "categories": [],
  "packageName": "@code-pushup/core",
  "plugins": [
    {
      "audits": [
        {
          "description": "Bundle size audit",
          "score": 1,
          "slug": "bundle-size",
          "title": "Bundle size",
          "value": 1,
        },
      ],
      "description": "Bundle size plugin for Code PushUp",
      "icon": "folder-css",
      "slug": "bundle-size",
      "title": "Bundle size",
    },
  ],
}
`;
